central/reports/validation/validate.go

```diff
@@ -324,3 +328,73 @@ func generateReportSnapshot(
 	snapshot.Notifiers = notifierSnaps
 	return snapshot
 }
+
+func generateViewBasedRequestName() string {
+	prefix := "View-Based-Report"
+	now := time.Now()
+	date := now.Format("Jan02")
+	year := now.Format("2006")
+	shortUUID := strings.Split(uuid.New().String(), "-")[0]
+	return fmt.Sprintf("%s-%s-%s-%s", prefix, strings.ToLower(date), year, shortUUID)
+}
+
+// ValidateAndGenerateViewBasedReportRequest validates a view-based report request and constructs the scheduler payload.
+func (v *Validator) ValidateAndGenerateViewBasedReportRequest(
+	req *apiV2.ReportRequestViewBased,
+	requesterID authn.Identity,
+) (*reportGen.ReportRequest, error) {
+	if req == nil {
+		return nil, errors.Wrap(errox.InvalidArgs, "Empty request")
+	}
+
+	// Currently only vulnerability view-based reports are supported.
+	if req.GetType() != apiV2.ReportRequestViewBased_VULNERABILITY {
+		return nil, errors.Wrap(errox.InvalidArgs, "unsupported report type")
+	}
+
+	// Validate filters.
+	vbFilters := req.GetViewBasedVulnReportFilters()
+	if vbFilters == nil {
+		return nil, errors.Wrap(errox.InvalidArgs, "view-based vulnerability report filters must be provided")
+	}
+	if len(vbFilters.GetImageTypes()) == 0 {
+		return nil, errors.Wrap(errox.InvalidArgs, "view-based report filters must specify image types")
+	}
+
+	// Convert API filters to storage filters.
+	storageFilters := &storage.ViewBasedVulnerabilityReportFilters{
```

Comment:
You might also want to add access scope rules so that the generated reports only include the workloads in the requestor's scope. See how it is done for on demand reports [here](https://github.com/stackrox/stackrox/blob/master/central/reports/validation/validate.go#L297)